docs/smart-contracts/example-contracts/workspace.mdx

```diff
@@ -0,0 +1,273 @@
+
+### Contract A: The Logic
+
+If the `contract_a_interface` crate is only present to define which functions, inputs, and outputs should be part of our contract, the `contract_a` crate is only present to actually code what the interface function(s) should actually _do_ to return the correct type of response.
+
+You can see our contract implementation of the `ContractAInterface` takes the value of `x`, along with the value of `y`, and performs a `checked_add`, returning the sum of both numbers (while avoiding an overflow error).
+
+:::info
+
+You may have noticed this contract doesn't require an `Env` argument. That's totally fine! `Env` has a ton of useful features that are available to your smart contracts, if you need them. But, you're not at all required to use it, if you don't.
+
+:::
+
+All that is required to make use of the previously defined `ContractAInterface` is to `use` the interface, define a `ContractA` struct, and then `impl` the interface on top of that struct.
+
+This crate uses the `contractimpl` attribute macro on the `ContractA` implementation, making the `add` function public and invocable by others on the Stellar network.
+
+### Contract B: The Invocation
+
+Now that we've created a trait in `contract_a_interface`, and implemented it in `contract_a`, we can use the `contract_b` crate to actually invoke the `add` function and get the sum of our integers.
+
+We're creating and implementing an entirely new contract, `ContractB`. We're skipping the trait, and getting right to the contract itself. We're making a function, `add_with`, which will run inside the Soroban environment. When it is invoked, it requires three arguments:
+
+- `contract: Address` - The address of a contract which implements that required client interface, `ContractAClient`.
+- `x: u32` and `y: u32` - The two numbers we want to (safely) compute the sum of.
+
+`ContractB` then invokes the `add` function from the deployed `ContractA`, returning its value back to the original invoker. It's a bit of a long round-trip for this simple example, but it illustrates a really powerful way you can separate out interface/trait definitions from contract logic in a way that allows crates to be reusable in a very powerful way.
+
+## Practical Use-Case Examples
+
+Outside the world of adding two integers together, this technique is far more versatile and useful in the real world. For example, this strategy could be used to:
+
+- Create and reference a standardized, consistent token interface?
+- Reuse a single interface that you want to incorporate across many different contracts?
+- Other things, too?
+
+## Build the Contracts
+
+To build the contract into a set of `.wasm` files, use the `stellar contract build` command. Both `workspace/contract_a` and `workspace/contract_b` must be built, and you can use a single command, since our workspace defines its `members` in the `Cargo.toml` file:
```

Comment:
```suggestion
To build the contracts into a set of `.wasm` files, use the `stellar contract build` command. Both `workspace/contract_a` and `workspace/contract_b` will be built, and you can use a single command, since our workspace defines its `members` in the `Cargo.toml` file:
```